pkg/kubelet/dockertools/labels.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockertools
+
+import (
+	"strconv"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// This file contains all docker label related constants and functions, including:
+//  * label setters and getters
+//  * label filters (maybe in the future)
+
+const (
+	kubernetesPodNameLabel      = "io.kubernetes.pod.name"
+	kubernetesPodNamespaceLabel = "io.kubernetes.pod.namespace"
+	kubernetesPodUID            = "io.kubernetes.pod.uid"
+
+	kubernetesPodLabel                    = "io.kubernetes.pod.data"
+	kubernetesTerminationGracePeriodLabel = "io.kubernetes.pod.terminationGracePeriod"
+	kubernetesContainerLabel              = "io.kubernetes.container.name"
+	kubernetesContainerRestartCountLabel  = "io.kubernetes.container.restartCount"
+)
+
+func newLabels(container *api.Container, pod *api.Pod, restartCount int) map[string]string {
+	// TODO (random-liu) Move more label initialization here
+	labels := map[string]string{}
+	labels[kubernetesPodNameLabel] = pod.Name
+	labels[kubernetesPodNamespaceLabel] = pod.Namespace
+	labels[kubernetesPodUID] = string(pod.UID)
+
+	labels[kubernetesContainerRestartCountLabel] = strconv.Itoa(restartCount)
+
+	return labels
+}
+
+// (random-liu) Where should the error be reported? Outside of the function or inside?
+func getRestartCountFromLabel(labels map[string]string) int {
+	if restartCountString, found := labels[kubernetesContainerRestartCountLabel]; found {
+		restartCount, err := strconv.Atoi(restartCountString)
+		if err != nil {
+			// This really should not happen, just in case
+			glog.Errorf("Parsing restart count string %s error: %v", restartCountString, err)
+			// Just set restartCount to 0 to handle this abnormal case
+			restartCount = 0
+		}
+		return restartCount
+	} else {
+		// Get restartCount from docker label. If there is no restart count label in an container,
+		// it should be an old container or an invalid container, we just set restart count to 0.
+		// Don't report error, because there must be many old containers without the label now
+		glog.V(3).Infof("Container doesn't have label %s, it may be an old or invalid container", kubernetesContainerRestartCountLabel)
```

Comment:
For the logging message to be meaningful in anyway, we need the container id. I'd suggest returning (int, error) and let the caller handle the error.